EIPS/eip-1014.md

```diff
@@ -0,0 +1,23 @@
+---
+eip: 1014
+title: Skinny CREATE2
+author: Vitalik Buterin (@vbuterin)
+category: Core
+type: Standards Track
+status: Draft
+created: 2018-04-20
+---
+
+### Specification
+
+Adds a new opcode at 0xf5, which takes 4 stack arguments: endowment, memory_start, memory_length, salt. Behaves identically to CREATE, except using `sha3(msg.sender ++ salt ++ init_code)[12:]` instead of the usual sender-and-nonce-hash as the address where the contract is initialized at.
+
+### Motivation
+
+Allows interactions to (actually or counterfactually in channels) be made with addresses that do not exist yet on-chain but can be relied on to only possibly eventually contain code that has been created by a particular piece of init code. Important for state-channel use cases that involve counterfactual interactions with contracts.
+
+#### Option 2
+
+Use `sha3(0xff ++ msg.sender ++ salt ++ init_code)[12:]`
```

Comment:
Reading from the comments it's:

``sha3(0xff ++ msg.sender ++ salt ++ sha3(init_code))[12:]``

instead of

``sha3(0xff ++ msg.sender ++ salt ++ init_code)[12:]``

so with the hash value ``sha3(init_code)`` and not the code itself. Am I correct on this?

Reply:
Please see https://github.com/ethereum/EIPs/pull/1375 . But yes, the `init_code`, not the code itself

Reply:
Ah, thanks!